### PR TITLE
arch/x86_64:Add some macro definitions

### PR DESCRIPTION
**Pull request by xianglyc**
## Summary
Add some macro definitions
## Impact
no impact
## Testing
ostest

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



**No, this PR does not meet the NuttX requirements.**

**Missing Information:**

* **Summary:** 
    * Lacks a clear explanation of why the macro definitions are necessary (fix, update, new feature). 
    * Does not specify which functional part of the code is being changed.
    * Omits details about how the change works and what exactly will change.
    *  No related issue references. 
* **Impact:** 
    *  While stating "no impact," it's crucial to address *all* impact categories with specifics. For instance:
        *  **Impact on build:** Even adding macros *might* affect the build process if they introduce new dependencies or change compilation flags. 
        * **Impact on documentation:** If these macros are user-facing, documentation would need updating.
* **Testing:**
    *  "ostest" is insufficient. Provide:
        *  Specific build host details (OS, CPU, compiler version).
        *  Target details (architecture, board, configuration).
        *  Relevant snippets from testing logs demonstrating the change's effect (both before and after). 

**Recommendations:**

1. **Expand the Summary:** Clearly articulate the purpose and functionality of the added macros.
2. **Address All Impact Categories:**  Thoroughly evaluate and describe the potential impact on all aspects, even if it seems minimal. 
3. **Provide Detailed Testing Information:** Include specific host/target configurations and relevant log snippets.

File: arch/x86_64/include/irq.h [hunk 1]
```diff
@@ -83,11 +83,19 @@ extern "C"
...
+ *   None
+ *
+ * Returned Value:
+ *   An integer index in the range of 0 through (CONFIG_SMP_NCPUS-1) that
+ *   corresponds to the currently executing CPU.
  *
  ****************************************************************************/
 
-#ifdef CONFIG_ARCH_HAVE_MULTICPU
+#ifdef CONFIG_SMP
```
Comment (xiaoxiang781216) [inline arch/x86_64/include/irq.h:98]: revert the change

File: arch/x86_64/include/irq.h [hunk 2]
```diff
@@ -100,7 +108,9 @@ static inline_function int up_cpu_index(void)
 
   return cpu;
 }
-#endif /* CONFIG_ARCH_HAVE_MULTICPU */
```
Comment (xiaoxiang781216) [inline arch/x86_64/include/irq.h:103]: ditto

File: arch/x86_64/include/intel64/irq.h [hunk 3]
```diff
@@ -461,6 +461,10 @@
 
 #define XMMAREA_REGS     (25)
 
+/* Aux register used by implementation */
```
Comment (raiden00pl) [inline arch/x86_64/include/intel64/irq.h:464]: please remove changes from in this file. These are not used in upstream yet

File: arch/x86_64/include/intel64/arch.h [hunk 4]
```diff
@@ -275,7 +277,7 @@
 #  define MSR_X2APIC_ICR_OTHERS        0x000c0000  /* Send to all APICs, excluding self. */
 #  define MSR_X2APIC_ICR_BUSY          0x00001000
 #  define MSR_X2APIC_ICR_FIXED         0x00000000
-#  define MSR_X2APIC_DESTINATION(d)    ((d) << 32)
+#  define MSR_X2APIC_DESTINATION(d)    ((d) << 32ul)
```
Comment (raiden00pl) [inline arch/x86_64/include/intel64/arch.h:280]: please remove this change, it was fixed by a8ec8b8cd291e6edd7be269cb02b420925588d49
